### PR TITLE
Put GetDirectoryName in try/catch

### PR DESCRIPTION
**Pull request by qinezh**
@herohua @vwxyzh @hellosnow @chenkennt

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs [hunk 1]
```diff
@@ -93,7 +93,18 @@ internal ManifestItem Transform(InternalManifestItem item)
...
                 string outputPath = Path.Combine(outputDirectory, outputFile);
-                var dir = Path.GetDirectoryName(outputPath);
+                string dir = null;
+                try
+                {
+                    dir = Path.GetDirectoryName(outputPath);
+                }
+                catch (PathTooLongException e)
+                {
+                    var message = $"Error getting the directory name for {outputPath} when process {item.LocalPathFromRoot}: {e.Message}";
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs]: >Error [](start = 36, length = 5)

Specified the exception name like "PathTooLongException occurs"?

File: src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs [hunk 2]
```diff
@@ -93,7 +93,18 @@ internal ManifestItem Transform(InternalManifestItem item)
...
+                    Logger.LogError(message);
+                    throw new PathTooLongException(message, e);
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs]: >PathTooLongException [](start = 30, length = 20)

question, why not transfer to DocumentException?

File: src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs (hunk 1 above)
Comment (qinezh) [inline src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs]: I think `{e.Message}` would indicate it

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**